pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +161,49 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+// True means that number of approvers is valid
+func (e *Executor) validateApproverNum(ctx context.Context, approver string, minApproverNum int) bool {
+	if minApproverNum <= 1 {
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approvedByKey, approver); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		e.LogPersister.Infof("Got approval from %q", approver)
+		e.reportApproved(approver)
+		e.LogPersister.Infof("This stage has been approved by %d user (%s)", minApproverNum, approver)
+		return true
+	}
+
+	const delimiter = ", "
+	as, _ := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	var approvedUsers []string
+	if as != "" {
+		approvedUsers = strings.Split(as, delimiter)
+	}
+
+	for _, u := range approvedUsers {
+		if u == approver {
+			e.LogPersister.Infof("Approval from the same user (%s) will not be counted", approver)
+			return false
+		}
+	}
+	approvedUsers = append(approvedUsers, approver)
+	e.LogPersister.Infof("Got approval from %q", approver)
+
+	aus := strings.Join(approvedUsers, delimiter)
+	if remain := minApproverNum - len(approvedUsers); remain > 0 {
+		e.LogPersister.Infof("Waiting for %d other approvers...", remain)
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, aus); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		return false
+	}
+
+	if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approvedByKey, aus); err != nil {
+		e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+	}
```

Comment:
How about this?

```suggestion
	if remain := minApproverNum - len(approvedUsers); remain > 0 {
		e.LogPersister.Infof("Waiting for %d other approvers...", remain)
		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approvedByKey, aus); err != nil {
			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
		}
		return false
	}
```

Reply:
In that case, I think the current approver doesn't save into metadata.

Reply:
@nghialv 

Is that intentional?

Reply:
Nice point. Let me send another suggestion to have it.

Reply:
@ono-max How about this one? 
https://github.com/pipe-cd/pipe/pull/2802/files#r756661077

Reply:
Great suggestion.
Thank you.